compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -962,8 +961,8 @@ checkFeature feature = do
 
 checkTemplateKey :: MonadGamma m => ExprVarName -> Qualified TypeConName -> TemplateKey -> m ()
 checkTemplateKey param tcon TemplateKey{..} = do
+    checkType tplKeyType KStar
```

Comment:
Unrelated change: the key type should be typechecked in the empty context. This matches the behavior of the scala typechecker and the spec.